### ISSUE TITLE
Try more efficient way to detect hero thumbnails?

### ISSUE THREAD

**Issue by Umiiii** [syurchen/underlords#2]
I've noticed that you are using OpenCV to detect every hero, but it seems slow and buggy.
Have you ever try another way to detect it? Such as machine learning or hash distance calculation.

**Comment by syurchen** [comment]
Yeah, detecting is not good. I will probably try something different in near future. Can you recommend examples I can follow or literature I can read on topic?

**Comment by umiiii** [comment]
https://developer.apple.com/documentation/createml/creating_an_image_classifier_model
Here is a simple example, and I tried with a little small dataset, so the model overfits, but got better performance.
![2](https://user-images.githubusercontent.com/29030474/64870961-de906200-d676-11e9-9f44-1c3ad31424ec.png)
(The failed one)
![1](https://user-images.githubusercontent.com/29030474/64870964-e3551600-d676-11e9-935d-f463a2bae15f.png)
(The successful one)

The dataset is transformed by GaussianBlur first, here is the reason:
![WX20190913-223838](https://user-images.githubusercontent.com/29030474/64871167-3d55db80-d677-11e9-99e7-056dd132702e.png)

**Comment by umiiii** [comment]
Please notice that there is no need to use the Apple ML framework. You may just train the classifier using another deep learning framework, I used because it is out-of-the-box.

**Comment by syurchen** [comment]
Thanks, I will try pytorch in separate branch. Let's see, if I will be able to come up with anything good
